src/utils/telescopeStates.js

```diff
@@ -6,15 +6,17 @@ export const getTelescopeState = async (site, telescope, enclosure) => {
   // This needs to be wrapped in a promise to ensure that the data (which is async) is returned and handled in the realTimeSessions store
   return new Promise((resolve, reject) => {
     fetchApiCall({
-      url: `${configurationStore.observationPortalUrl}telescope_states/?site=${site}&telescope=${telescope}`,
+      url: `${configurationStore.observationPortalUrl}telescope_states/?site=${site}&telescope=${telescope}&enclosure=${enclosure}`,
       method: 'GET',
       successCallback: (response) => {
-        const data = response && response[`${site}.${enclosure}.${telescope}`][0]
+        let data
+        if (!response || !response[`${site}.${enclosure}.${telescope}`]) {
+          // Sometimes there is no response or the response is empty. In this case, we want the user to keep going with their rti session
+          data = 'Reason unavailable'
+        } else {
+          data = response && response[`${site}.${enclosure}.${telescope}`][0]
```

Comment:
This returns a dictionary, not a string like your error case returns. I think you could pull out `event_reason` if you want it to return a reason string.